### PR TITLE
fix breakage in FM tests

### PR DESCRIPTION
**Pull request by chliangGoogle**
The unit test is using a different path of plist file than the production code causing the test failed as it is checking the path of production code instead of unit test path. This fixes [#7797](https://github.com/firebase/firebase-ios-sdk/issues/7797)

### CODE REVIEW

**Comment by paulb777** [comment]
Thanks! Looks like there's still a failure in the spm tests.

Also, in case you missed it, there's a 2nd problem described in #7797 about header specifications causing the prerelease nightly test to fail.

File: FirebaseMessaging/Tests/UnitTests/FIRMessagingCheckinStoreTest.m
Comment (morganchen12) [inline FirebaseMessaging/Tests/UnitTests/FIRMessagingCheckinStoreTest.m:76]: Optionally, you can pass in a nonnull error pointer here and validate that the delete succeeds in tests.

File: FirebaseMessaging/Tests/UnitTests/FIRMessagingCheckinStoreTest.m
Comment (morganchen12) [inline FirebaseMessaging/Tests/UnitTests/FIRMessagingCheckinStoreTest.m:96]: nit: use `self.checkinStore` to be consistent with access above

File: FirebaseMessaging/Tests/UnitTests/FIRMessagingCheckinStoreTest.m
Comment (morganchen12) [inline FirebaseMessaging/Tests/UnitTests/FIRMessagingCheckinStoreTest.m:125]: `self.checkinStore`

File: FirebaseMessaging/Tests/UnitTests/FIRMessagingCheckinStoreTest.m
Comment (morganchen12) [inline FirebaseMessaging/Tests/UnitTests/FIRMessagingCheckinStoreTest.m:153]: `self.checkinStore`

File: FirebaseMessaging/Tests/UnitTests/FIRMessagingCheckinStoreTest.m
Comment (morganchen12) [inline FirebaseMessaging/Tests/UnitTests/FIRMessagingCheckinStoreTest.m:176]: `self.checkinStore`

**Review by morganchen12 — APPROVED** [review]
LGTM, but please wait for other approvers before merging.

File: FirebaseMessaging/Tests/UnitTests/FIRMessagingCheckinStoreTest.m
Comment (chliangGoogle) [reply FirebaseMessaging/Tests/UnitTests/FIRMessagingCheckinStoreTest.m:96]: This is to avoid implicitly retain 'self' warning: https://stackoverflow.com/questions/21577711/block-implicitly-retains-self-explicitly-mention-self-to-indicate-this-is-i?noredirect=1&lq=1

File: FirebaseMessaging/Tests/UnitTests/FIRMessagingCheckinStoreTest.m
Comment (chliangGoogle) [reply FirebaseMessaging/Tests/UnitTests/FIRMessagingCheckinStoreTest.m:76]: This is in teardown, I don't usually assert error there but i might be wrong.

**Review by paulb777 — APPROVED** [review]
LGTM on green (except travis which looks broken right now)

**Comment by chliangGoogle** [comment]
@paulb777 weirdly messaging is one of the test in travis that is green. It doesn't look like it's relavant.

**Review by christibbs — APPROVED** [review]

**Pull request merged**
